doc/source/numpy_2_0_migration_guide.rst

```diff
@@ -92,27 +92,29 @@ geterrobj               Use the np.errstate context manager instead.
 Inf                     Use ``np.inf`` instead.
 Infinity                Use ``np.inf`` instead.
 infty                   Use ``np.inf`` instead.
-issctype
+issctype                Use ``isinstance(rep, (type, np.dtype))`` instead.
```

Comment:
```suggestion
issctype                Use ``isinstance(rep, (np.dtype.type, np.dtype))`` instead.
```

I think this is actually the equivalent thing

Reply:
I took `isinstance(rep, (type, np.dtype))` from `def issctype` in: https://github.com/numpy/numpy/blob/a555e7dfcb6e4a631287aaf8661d3f225816faf8/numpy/_core/numerictypes.py#L221
where `type` is a builtin. 

Also, in repl I see that `type(np.dtype.type)` is just `TypeNone`. WDYT?

Reply:
Ah, you're right of course that my suggestion doesn't make sense. What I was going for is that the suggested check (what the implementation in `issctype` does) isn't useful, since then *any* type is a scalar type:

```
In [4]: isinstance(dict, (np.dtype, type))
Out[4]: True
```

Maybe better to say that the check it did wasn't useful and there isn't a migration?

Reply:
I would still give a migration that would be somewhat close to the original function. How about `issubclass(rep, np.generic)`? 
It yields True for scalar types, and False for e.g. Python builtins. For `np.dtype(...)` passed as the first argument it raises an exception `issubclass() arg 1 must be a class`, so user notices that `.type` must be added. 
(A similar check is present in `def obj2sctype`)

Reply:
Sure, that sounds more useful than the oroginal suggestion.

Reply:
Ok - I updated it.